### DISCUSSION TITLE
Meeting 1 - Meeting Notes

### DISCUSSION THREAD

**Discussion by AdamFrisby** [omigroup/omi-scripting-group#9] [Announcements]
### Meeting 1 (Introductions)
**Held:** Tuesday 11th January 2022 at 1700 UTC.
**In Attendence:** Adam, A., antpb, Cafezinho, cdata, humbletim, jeana, jin, Koolala, Lyuma, Meticulac, mrmetaverse, Rick W, Robert, SafariMonkey. (and others who may have joined later)

### Agenda
- Introductions: Participants were invited to introduce themselves and their expertise.
- Open Floor
    - Adam:
        - Discussed personal goals - standardising ABI & API
        - Prior experience 25+yrs in virtual worlds (most professionally). Happy to contribute knowledge from previous projects and expertise
        - Direct experience implementing scripting runtimes and APIs in virtual worlds
    - Robert:
        - Interested in the API available, as well as a performance focus to allow e.g. Javascript to be embedded within these applications and creating a Scene Graph API for Interactive 3D Scenes & Content
    - Cdata:
        - Interested in capturing widest audience and list of use cases
    - Lyuma
        - Interested in building building proof of concepts and starting small - experimenting first.
    - Koolala:
        - Less interested in standardising the ABI/API, and has some concerns about ways this can be monetised (e.g. NFT projects).
    - humbletim:
        - Interested in bringing in other stakeholders, and a focus on sandboxing and security
        - Webpages don’t allow random visitors to bring in scripting - but this is what we may be enabling, bringing foreign content into ‘webpages’ is quite difficult
        - Wants to balance utility and security
    - Wider group discussion about sandboxing
        - API limitations
        - Memory use
    - cdata:
        - Suggestion to do a review of wasm (Adam: seconded, volunteering)
    - Robert:
        - Process discussion on the importance of a contributor agreement when contributing to the specification (either samples, or just technical recommendations). Need to be careful of the W3C Rules.
    - Rick W:
        - Interested in contributing feedback as a potential user of the specification
    - Wider group discussion of Scope
        - Adam: interested in specifying “the widest possible array” of use cases, enabling any 3D application from games to serious applications to use the functionality.
        - Adam: Suggestion to look at Game Engine APIs as a good starting point - but obviously not to crib directly from.
    - Group discussion of timeline
        - Should we pre-empt the large tech majors?
    - Group discussion of Exploration
        - It was encouraged for participants to “go out and experiment!”
        - Also suggested it is worth looking extensively at past platforms
    - Rick W:
        - RP1 Good inspiration - the idea of jumping from game to game to school, etc.
            - Holy Grail - one entry point, all environments
    - SafariMonkey:
        - Consellation Project / Interop between whole engines
    - Discussion of Engine Neutrality
- Meeting Regularity
    - VOTE:
        - Tuesdays at 1600 UTC: 6 Votes For (Adam, Robert, mrmetaverse, SafariMonkey, A. Mike)
        - Tuesdays at 1700 UTC: 4 Votes For (Koolala, mrmetaverse, Rick W, cdata)
        - Tuesdays at 1800 UTC: 0 Votes
    - Proposal to alternate substantive working group meetings with knowledge and demo sessions on alternate weeks:
        - PASS: 9 Votes to 0, 3 Abstain (in favor: Adam, RickW, SafariMonkey, Koolala, Robert, cdata, antpb, jeana, Lyuma)
        - First meeting next week at 1600 UTC (18th Jan 2022) - Introduction to WASM
- Retrospectives
    - Discussed but deferred for a future meeting

**Next Meeting:** Tuesday 18th January 2022 - Workshop & Demo Meeting: Introduction to WASM
**Location:** Discord

**Comment by AdamFrisby** [comment]
Record of votes & attendance:
![Screenshot_13](https://user-images.githubusercontent.com/114041/149000021-8c38cd9f-a7e9-4fd8-a007-2892d2c6a31c.png)
![Screenshot_12](https://user-images.githubusercontent.com/114041/149000025-9d17ecce-7fff-42ae-a526-6ea0e60ef8ef.png)
![Screenshot_9](https://user-images.githubusercontent.com/114041/149000029-1b1f0c82-886a-4aaf-b187-62b75c131e35.png)
![Screenshot_11](https://user-images.githubusercontent.com/114041/149000031-a0b3eea4-07b6-4192-b871-8e4a542d8e65.png)
![Screenshot_10](https://user-images.githubusercontent.com/114041/149000033-18354ae0-f293-43b1-9c2c-a00aa57620c7.png)

**Comment by robertlong** [comment]
Here's my notes as well:

Adam
- Key items to work on:
  - WASM Sandbox
  - WASM SDK with standardized API
- Goal to get things out in the open for experimentation soon
Robert
- Would like to see a standard scene graph API
- High level API for integration into other engines
- With good performance
cdata
- Make our work achievable (head in the clouds but grounded in what we can achieve
- Need to appeal to others and bring them into this group (edited)
Lyuma
- Achieve a standard ABI
Koolala
- Doesn't believe we need a standard ABI or SDK
- We need a standard format for interacting with WASM in VR
- Concerns around NFTs
- Does think that we should continue working on an ABI or SDK if that's what we're looking for in this group (edited)
Humbletim
- Would like to be careful and consider security (edited)
* Discussion about UGC content security and moderation *
* Discussion about running this group under W3C for IP protection *
Rick W
- Interested in integrating the work of this group into their platform
Adam
- Hoping that this ABI can be general enough to apply to any 3D application
Safari Monkey
- What is the intended scope of that?
cdata
- Need to consider if that's feasible.
- Is it going to be like a general runtime for the metaverse or business logic for avatars? (edited)
jeana
- What's the turnaround rate for this project? When can we expect the work?
Adam
- Realistically up to 2 years
- But needs foundational layers to built up
cdata
- Need to think about proprietary SDKs
- Encourage this group to experiment before consensus (with agreement from Lyuma)
Adam
- Encourage people in this group to go experiment with other platforms and see what others have done in the past
* Discussion around meeting cadence *
Possibly alternating meetings with a working session one week and a retrospective / high level overview / demo session meeting?
People volunteering for giving presentations on other projects.

Vote on new meeting time 16:00 UTC (8:00 AM PST) passes 6-4

VOTE: Proposal to alternate substantive working group meetings with demo and knowledge sharing sessions on alternate weeks passes 9-0 with 2 abstentions
